elements/pl-truth-table/pl-truth-table.py

```diff
@@ -0,0 +1,567 @@
+                    "html": true_label if answer == "True" else false_label,
+                }
+
+                answer_set.append(answer_html)
+
+            row_set = []
+
+            for i, row in enumerate(display_rows):
+                student_answer = -1
+
+                correct_answer = correct_answers[i]
+
+                value_set = []
+
+                for j, val in enumerate(row["values"]):
+                    value_html = {"html": true_label if val == 1 else false_label}
+
+                    value_set.append(value_html)
+
+                correct_answer_set = []
+
+                for j in range(num_ans_columns):
+                    correct_answer = correct_answers[(i * num_ans_columns) + j]
+
+                    correct_answer_html = {
+                        "html": true_label if correct_answer == "True" else false_label
+                    }
+
+                    correct_answer_set.append(correct_answer_html)
+
+                row_html = {
+                    "key": row["key"],
+                    "values": value_set,
+                    "correct_answers": correct_answer_set,
+                }
+
+                row_set.append(row_html)
+
+            html_params = {
+                "answer": True,
+                "uuid": pl.get_uuid(),
+                "inputs": variable_set,
+                "answer_columns": ans_column_set,
+                "rows": row_set,
+            }
+
+            with open("pl-truth-table.mustache", "r", encoding="utf-8") as f:
+                html = chevron.render(f, html_params).strip()
+
+    return html
+
+
+def grade(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, "answers-name")
+    weight = pl.get_integer_attrib(element, "weight", WEIGHT_DEFAULT)
+    partial_credit = pl.get_boolean_attrib(
+        element, "partial-credit", PARTIAL_CREDIT_DEFAULT
+    )
+    display_variables, display_rows, display_ans_columns = data["params"][name]
+
+    submitted_answers = data["submitted_answers"].get(name, [])
+    correct_answers = data["correct_answers"].get(name, [])
+    number_answers = len(correct_answers)
+
+    # Count the number of answers that are correct.
+    num_correct = 0
+    for i in range(0, number_answers):
+        student_answer = submitted_answers[i]
+
+        if student_answer == correct_answers[i]:
+            num_correct += 1
+
+    score = 0
+    if not partial_credit and num_correct == number_answers:
+        score = 1
+    elif partial_credit:
+        # EDC grading
+        score = num_correct / number_answers
+    data["partial_scores"][name] = {"score": score, "weight": weight}
+
+
+def test(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, "answers-name")
+    weight = pl.get_integer_attrib(element, "weight", WEIGHT_DEFAULT)
+
+    _, display_options = data["params"][name]
```

Comment:
Great to see you've implemented the test function here! Add a line to `tests/exampleCourseQuestions.test.js` so that the test will be run automatically as part of the continuous integration pipeline we have set up.